### PR TITLE
Add CI support using Travis CI

### PR DESCRIPTION
**Pull request by ePirat**
Add a basic `.travis-ci.yml` that builds MoltenVK

### CODE REVIEW

**Comment by billhollings** [comment]
Thanks, Karl.

I've added issue #75 to remember for the badge and to contact Khronos webmaster.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+# macOS and Xcode Version
+# Xcode 9.2 running on macOS 10.12
+osx_image: xcode9.2
+
+# Build dependencies
+install:
+  - brew install python3
+  - sudo gem install asciidoctor
+  - ( cd External && ./makeAll )
```
Comment (billhollings) [inline .travis.yml]: Will this ensure the submodules are pulled in?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# Cache built deps
+cache:
+  directories:
+    - External
+
+script:
+  - xcodebuild -scheme "MoltenVK (Debug)"
```
Comment (billhollings) [inline .travis.yml]: Probably better to build the Release scheme?

Comment (ePirat) [reply .travis.yml]: I wasn't sure, while release is what most people build, there could be mistakes in code that is only enabled in Debug mode, which would then slip through the CI tests.

File: .travis.yml (hunk 1 above)
Comment (ePirat) [reply .travis.yml]: Travis automatically checks out the repo with the submodules.

File: .travis.yml (hunk 2 above)
Comment (billhollings) [inline .travis.yml]: @ePirat 

Can you also add the Travis badge to the README as @karl-lunarg recommends, and tracked in issue #75, please? Thanks.

Comment (ePirat) [reply .travis.yml]: Sure!

**Comment by ePirat** [comment]
Updated PR to include a commit which adds the Travis CI badge to the Readme.

File: .travis.yml (hunk 2 above)
Comment (billhollings) [reply .travis.yml]: That's a fair point. I guess Debug mode does give us the best coverage. I'm happy to leave it as Debug mode then.

**Pull request merged**